app/models/User.coffee

```diff
@@ -587,7 +587,7 @@ module.exports = class User extends CocoModel
   allowStudentHeroPurchase: -> features?.classroomItems ? false and @isStudent()
   canBuyGems: -> not (features?.chinaUx ? false)
   constrainHeroHealth: -> features?.classroomItems ? false and @isStudent()
-  promptForClassroomSignup: -> not (features?.chinaUx ? false or window.serverConfig?.codeNinjas ? false or features?.brainPop ? false)
+  promptForClassroomSignup: -> not ((features?.chinaUx ? false) or (window.serverConfig?.codeNinjas ? false) or (features?.brainPop ? false))
   showAvatarOnStudentDashboard: -> not (features?.classroomItems ? false) and @isStudent()
```

Comment:
Out of interest because you just looked at this. How does the precedence work with `and`?
This `showAvatarOnStudentDashboard` has brackets, but the `showGearRestrictionsInClassroom` beneath doesn't have brackets. Don't need any actions. I am just interested.

Reply:
Yes, `and` also seems to have higher precedence than `?` but did not test it with an example.